muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/general/user-language-profile/index.tsx

```diff
@@ -0,0 +1,391 @@
+                  defaultValue={futureUsage || ""}
+                  className="user-language-profile__textarea"
+                />
+              </div>
+            )}
+            {skillGoals && (
+              <div className="user-language-profile-container__row">
+                <label
+                  htmlFor="skillGoals"
+                  className="user-language-profile__label"
+                >
+                  {t("labels.languageGoals", {
+                    ns: "languageProfile",
+                  })}
+                </label>
+                <div className="language-profile__field-description">
+                  {t("content.languageGoals", {
+                    ns: "languageProfile",
+                  })}
+                </div>
+                <textarea
+                  id="skillGoals"
+                  disabled={true}
+                  defaultValue={skillGoals || ""}
+                  className="user-language-profile__textarea"
+                />
+              </div>
+            )}
+          </Accordion>
+          {languages?.map((lang) => (
+            <Accordion
+              id={`language-drawer-${lang.code}`}
+              key={lang.code}
+              title={lang.name}
+            >
+              {lang.levels ? (
+                lang.levels.map((level, idx) => (
+                  <div key={idx}>
+                    {Object.entries(level).map(([level, value]) => {
+                      // Extracting the context index from the skill string
+                      // the skill keys are in the format "fi-0", "fi-1", etc.
+                      // This is because how the data handling component works
+
+                      const contextIndex = Number(level.split("-")[1]);
+                      return (
+                        <>
+                          <h4>
+                            {t("labels.languageSkillLevel", {
+                              ns: "languageProfile",
+                              context: levelMap[contextIndex],
+                            })}
+                          </h4>
+
+                          <div
+                            className="user-language-profile__drawer-skill"
+                            key={level}
+                          >
+                            <h5>
+                              {t([`levels.${value}`, "levels.unknown"], {
+                                ns: "languageProfile",
+                                defaultValue: "Unknown",
+                              })}
+                            </h5>
+                            <p>
+                              {t([`levels.${value}`, "levels.unknown"], {
+                                ns: "languageProfile",
+                                context: levelMap[contextIndex],
+                                defaultValue: "Unknown",
+                              })}
+                            </p>
+                          </div>
+                        </>
```

Comment:
Probably should check if browser console is giving error as element that has key defined doesn't include h4 title. If so change fragments to <React.Fragment key={level}> and move key to that wrapper